File: TuduManayer/java/TuduManager/web.api.spring.boot/src/test/java/net/seymourpoler/web/api/spring/boot/todo/SearchTodoControllerShould.java
package net.seymourpoler.web.api.spring.boot.todo;

import net.seymourpoler.domain.todo.search.ISearchTodoService;
import net.seymourpoler.domain.todo.search.models.Todo;
import org.junit.Test;
import org.springframework.http.HttpStatus;

import java.util.List;

import static org.assertj.core.api.AssertionsForInterfaceTypes.assertThat;
import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.when;

public class SearchTodoControllerShould {

    @Test
    public void return_todos() {
        var searchTodoService = mock(ISearchTodoService.class);
        final String searchText = "search text";
        var todos = List.of(new Todo());
        when(searchTodoService.search(searchText)).thenReturn(todos);
        var controller = new SearchTodoController(searchTodoService);

        var result = controller.seach(searchText);

        assertThat(result.getStatusCode()).isEqualTo(HttpStatus.OK);
        assertThat(result.getBody()).isEqualTo(todos);
    }
}

File: TuduManayer/dotnet/react.mvc-core.postgres.dapper/TuduManager/test/TuduManayer.web.api.Test/Todo/Search/SearchTodoControllerShould.cs
using System.Collections.Generic;
using System.Net;
using Microsoft.AspNetCore.Mvc;
using Moq;
using Shouldly;
using TuduManayer.Domain.Todo.Search;
using TuduManayer.web.api.Todo.Search;
using Xunit;

namespace TuduManayer.web.api.Test.Todo.Search
{
    public class SearchTodoControllerShould
    {
        private Mock<ISearchTodoService> service;
        private SearchTodoController controller;


        public SearchTodoControllerShould()
        {
            service = new Mock<ISearchTodoService>();
            controller = new SearchTodoController(service.Object);
        }

        [Fact]
        public void return_todos()
        {
            const string searchText = "search-text";
            var todos = new Domain.Todo.Search.Models.Todo[] { };
            service
                .Setup(x => x.Search(searchText))
                .Returns(todos);

            var response = controller.Search(searchText)  as OkObjectResult;
            
            response.StatusCode.Value.ShouldBe<int>((int)HttpStatusCode.OK);
            response.Value.ShouldBeOfType<Domain.Todo.Search.Models.Todo[]>();
        }
    }
}